mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/work/impl/PojoScopeWorkspaceImpl.java

```diff
@@ -28,23 +29,23 @@ public PojoScopeWorkspaceImpl(Set<? extends PojoWorkIndexedTypeContext<?, ?>> ta
 	}
 
 	@Override
-	public CompletableFuture<?> mergeSegments() {
-		return doOperationOnTypes( IndexWorkspace::mergeSegments );
+	public CompletableFuture<?> mergeSegments(OperationSubmitter operationSubmitter) {
+		return doOperationOnTypes( indexWorkspace -> indexWorkspace.mergeSegments( operationSubmitter ) );
```

Comment:
I hear capturing lambdas have a performance impact, so it would probably be better to pass the submitter to `doOperationOnTypes`, and turn the `Function` to a `BiFunction`, so you can write `doOperationOnTypes( IndexWorkspace::mergeSegments, operationSubmitter )`.

Reply:
Yeah. This was primarily because of the `#purge()` that was already using lambda to address the routing keys. changed 😃